node-commons/src/main/java/org/apache/iotdb/commons/path/fa/dfa/graph/NFAGraph.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.commons.path.fa.dfa.graph;
+
+import org.apache.iotdb.commons.conf.IoTDBConstant;
+import org.apache.iotdb.commons.path.PartialPath;
+import org.apache.iotdb.commons.path.fa.IFAState;
+import org.apache.iotdb.commons.path.fa.IFATransition;
+import org.apache.iotdb.commons.path.fa.dfa.DFAState;
+import org.apache.iotdb.commons.utils.TestOnly;
+
+import org.apache.commons.lang.StringUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class NFAGraph {
+  private final List<IFAState> nfaStateList = new ArrayList<>();
+
+  // [transitionIndex][stateIndex]List<IFAState>
+  private final List<IFAState>[][] nfaTransitionTable;
+
+  public NFAGraph(
+      PartialPath pathPattern, boolean isPrefix, Map<String, IFATransition> transitionMap) {
+    nfaTransitionTable = new List[transitionMap.size()][pathPattern.getNodeLength() + 1];
+    // init start state, curNodeIndex=0
+    int curStateIndex = 0;
```

Comment:
Give an example of what will the NFAGraph be like when input pattern xxxxxx.

Reply:
add Demo in PatternDFATest#printFASketch